kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/utils/qaseIO/TestCaseGenerator.java

```diff
@@ -10,52 +12,142 @@
     private static final CasesApi casesApi = new CasesApi(apiClient);
 
     @SneakyThrows
     public static void createTestCaseIfNotExists(Method testMethod) {
-
         TestCaseCreate caseCreate = new TestCaseCreate();
-        HashMap<Long, String> cases = getTestCasesTitleAndId();
-        String testCaseTitle = getClassName(MethodSource.from(testMethod)) + "." + testMethod.getName() + " : " +
-                MethodNameUtils.formatTestCaseTitle(testMethod.getName());
-
-        if (!cases.containsValue(testCaseTitle)) {
-            caseCreate.setTitle(testCaseTitle);
-            if (testMethod.isAnnotationPresent(Suite.class)) {
-                long suiteId = testMethod.getAnnotation(Suite.class).suiteId();
-                caseCreate.suiteId(suiteId);
+        String testCaseTitle = generateTestCaseTitle(testMethod);
+        if (!isMethodAnnotatedWithCaseId(testMethod) || !TestCaseGenerator.isCaseIdPresentInQaseIo(testMethod)) {
+            if (!isCaseTitleExistInQaseIo(testMethod)) {
+                caseCreate.setTitle(testCaseTitle);
+                caseCreate.setAutomation(getAutomationStatus(testMethod));
+                if (isMethodAnnotatedWithSuite(testMethod)) {
+                    long suiteId = testMethod.getAnnotation(Suite.class).suiteId();
+                    caseCreate.suiteId(suiteId);
+                }
+                Long caseId = Objects.requireNonNull(casesApi.createCase(getConfig().projectCode(), caseCreate).getResult()).getId();
+                log.info("New test case = '" + testCaseTitle + "' created with id " + caseId);
             }
-            casesApi.createCase(getConfig().projectCode(), caseCreate);
-            log.info("New test case = '" + testCaseTitle + "' created");
         }
     }
 
     @SneakyThrows
     public static HashMap<Long, String> getTestCasesTitleAndId() {
         HashMap<Long, String> map = new HashMap<>();
-        TestCaseListResponse response =
-                casesApi.getCases(getConfig().projectCode(), new Filters().status(Filters.SERIALIZED_NAME_STATUS), 100, 0);
-        for (int i = 0; i < Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size(); i++) {
-            map.put(response.getResult().getEntities().get(i).getId(),
-                    response.getResult().getEntities().get(i).getTitle());
+        boolean getCases = true;
+        int offSet = 0;
+        while (getCases) {
+            getCases = false;
+            TestCaseListResponse response =
+                    casesApi.getCases(getConfig().projectCode(), new Filters().status(Filters.SERIALIZED_NAME_STATUS), 100, offSet);
+            if (Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size() > 0) {
+                for (int i = 0; i < Objects.requireNonNull(Objects.requireNonNull(response.getResult()).getEntities()).size(); i++) {
+                    map.put(response.getResult().getEntities().get(i).getId(),
```

Comment:
`response.getResult().getEntities().get(i)` can be extracted into a variable

Reply:
done